MeterianHQ/jenkins-plugin
# [WIP] Create pull request on GitHub

- Applied changes to both the Simple build jobs and Multi stage build jobs in Jenkins.
- Uses the GitHub token applied to the meterian plugin via Jenkins config
- Manual tests performed
- Automated tests pending to be written

We currently do not check for these:

- if a branch already exists with the same name 
   - would you want to ignore this, give a warning and proceed to the PR creation stage?
   - would you want to throw an error and stop process?
   - ~would you want to ignore the existence and stop processing?~ (we do this currently if a branch exists on local or remote repos)
- if a PR already exists from the same Branch
      - would you want to give a warning and proceed to the PR creation?
      - would you want to throw an error and stop process?
      - would you want to ignore the existence and stop processing?
- if the fix has been already applied via another branch or PR (contents check)
      - We can give a warning that it already exists and nothing is done
- ~there is a chance that the Jenkins multistage builds might keep creating PRs against a branch which is already being fixed by meterian (recursive) - there's more than one way to handle this which one do you prefer?~ (fixed now, with above fix)

What are your thoughts on that?

We haven't written automated tests yet but that will follow once we are free from two runtime exceptions that only occur when running inside the jenkins session (see above) and once we have decided what we would like to do with the edge cases.
